### PR TITLE
Delivery guards

### PR DESCRIPTION
**Pull request by ioaNikas**
- [x] DeliveryActiveGuard and DeliveryListGuard now set the store
- [x] Movie guards now got stakeholders in state
- [x] Cleaned and refactored code for more consistency

### CODE REVIEW

File: libs/material/src/lib/delivery/delivery-view/delivery-view.component.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ export class DeliveryViewComponent implements OnInit, OnDestroy {
   }
 
   public editDelivery() {
-    this.router.navigate([`layout/${this.movie.id}/form/${this.query.getActiveId()}`]);
+    this.router.navigate([`layout/${this.movie.id}/${this.query.getActiveId()}/edit`]);
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/delivery/delivery-view/delivery-view.component.ts]: Not super readable...

File: libs/material/src/lib/delivery/guards/delivery-list.guard.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { StateListGuard, FireQuery, Query } from '@blockframes/utils';
+import { DeliveryStore, Delivery, modifyTimestampToDate, DeliveryDB } from '../+state';
+import { switchMap, map } from 'rxjs/operators';
+import { MovieQuery, Stakeholder } from '@blockframes/movie';
+import { Organization } from '@blockframes/organization';
+
+const deliveryQuery = (movieId: string): Query<DeliveryDB[]> => ({
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/delivery/guards/delivery-list.guard.ts]: ❤️

File: libs/material/src/lib/template/template-picker/template-picker.component.ts [hunk 3]
```diff
@@ -54,14 +54,14 @@ export class TemplatePickerComponent implements OnInit, OnDestroy {
...
-    this.router.navigate([`layout/${movieId}/form/${this.deliveryQuery.getActiveId()}/settings`]);
+    this.router.navigate([`layout/${movieId}/${this.deliveryQuery.getActiveId()}/settings`]);
     this.close();
   }
 
   public async useMovieAsTemplate(){
     this.close();
     await this.deliveryService.addMovieMaterialsDelivery();
-    this.router.navigate([`layout/${this.movieQuery.getActiveId()}/form/${this.deliveryQuery.getActiveId()}/settings`]);
+    this.router.navigate([`layout/${this.movieQuery.getActiveId()}/${this.deliveryQuery.getActiveId()}/settings`]);
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/template/template-picker/template-picker.component.ts]: Not readable

File: libs/material/src/lib/template/template-picker/template-picker.component.ts [hunk 4]
```diff
@@ -54,14 +54,14 @@ export class TemplatePickerComponent implements OnInit, OnDestroy {
     }
 
     this.deliveryService.addDelivery(templateId);
-    this.router.navigate([`layout/${movieId}/form/${this.deliveryQuery.getActiveId()}/settings`]);
+    this.router.navigate([`layout/${movieId}/${this.deliveryQuery.getActiveId()}/settings`]);
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/template/template-picker/template-picker.component.ts]: Not readable

File: libs/movie/src/lib/movie/+state/movie.model.ts [hunk 5]
```diff
@@ -1,10 +1,11 @@
 import { Organization } from "@blockframes/organization";
 import { Material } from "@blockframes/material";
+import { Stakeholder } from "@blockframes/movie/stakeholder/+state";
 
 export interface Movie {
   id: string,
   org?: Organization,
-  title: any, // will contain all titles: original, international, suiss, etc 
+  title: any, // will contain all titles: original, international, suiss, etc
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/+state/movie.model.ts]: Should be typed.

File: libs/movie/src/lib/movie/guards/movie-active.guard.ts [hunk 6]
```diff
@@ -2,9 +2,17 @@ import { Injectable } from '@angular/core';
 import { StateActiveGuard, FireQuery, Query } from '@blockframes/utils';
 import { Movie, MovieStore } from '../+state';
 import { Router } from '@angular/router';
+import { Stakeholder } from '@blockframes/movie/stakeholder/+state';
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/guards/movie-active.guard.ts]: Don't import with the alias "@blockframes" when you're in the same library.

File: libs/material/src/lib/delivery/guards/delivery-active.guard.ts [hunk 7]
```diff
@@ -3,9 +3,17 @@ import { StateActiveGuard, FireQuery, Query } from '@blockframes/utils';
...
 
 export const deliveryActiveQuery = (deliveryId: string): Query<DeliveryDB> => ({
-  path: `deliveries/${deliveryId}`
+  path: `deliveries/${deliveryId}`,
+  stakeholders: (delivery: DeliveryDB): Query<Stakeholder> => ({
+    path: `deliveries/${delivery.id}/stakeholders`,
+    organization: (stakeholder: Stakeholder): Query<Organization> => ({
+      path: `orgs/${stakeholder.orgId}`
+    })
+  })
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/delivery/guards/delivery-active.guard.ts]: You should not need to specify the subtypes : 
```typescript
export const deliveryActiveQuery = (deliveryId: string): Query<DeliveryDB> => ({
  path: `deliveries/${deliveryId}`,
  stakeholders: delivery => ({
    path: `deliveries/${delivery.id}/stakeholders`,
    organization: stakeholder => ({
      path: `orgs/${stakeholder.orgId}`
    })
  })
});
```
But you can keep them if you think it's clearer.

File: libs/material/src/lib/template/template-picker/template-picker.component.ts [hunk 8]
```diff
@@ -46,22 +45,20 @@ export class TemplatePickerComponent implements OnInit, OnDestroy {
     this.templates$ = this.query.selectAll();
   }
 
-  public createDelivery(templateId?: string, ) {
-    const movieId = this.movieQuery.getActiveId();
-
+  public async createDelivery(templateId?: string, ) {
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/template/template-picker/template-picker.component.ts]: Remove the "," in the parameter

File: libs/material/src/lib/delivery/delivery-view/delivery-view.component.ts [hunk 9]
```diff
@@ -63,10 +61,6 @@ export class DeliveryViewComponent implements OnInit, OnDestroy {
     this.allChecked = false;
   }
 
-  public editDelivery() {
-    this.router.navigate([`layout/${this.movie.id}/${this.query.getActiveId()}/edit`]);
-  }
-
   ngOnDestroy() {
     this.isAlive = false;
```
[the remark this answers is not in the document]
Comment (ioaNikas) [reply libs/material/src/lib/delivery/delivery-view/delivery-view.component.ts]: File is not shown completly on this PR, we still subscribe on materials in this component. This should be removed in next PR with material guards :+1:

**Pull request merged**